From 4d34a8286295d1eef2be24959fc51249b4322ec1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Dan=20Vr=C3=A1til?= <dvratil@redhat.com>
Date: Mon, 31 Mar 2014 14:17:07 +0200
Subject: [PATCH 24/48] Catch Xapian::DatabaseException in EmailQuery

Prevents KMail from crashing when using quick filter when Baloo databases
are missing.
---
 src/pim/lib/emailquery.cpp | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/pim/lib/emailquery.cpp b/src/pim/lib/emailquery.cpp
index f4710ff..2964846 100644
--- a/src/pim/lib/emailquery.cpp
+++ b/src/pim/lib/emailquery.cpp
@@ -182,7 +182,13 @@ void EmailQuery::setRead(bool read)
 ResultIterator EmailQuery::exec()
 {
     const QString dir = KGlobal::dirs()->localxdgdatadir() + "baloo/email/";
-    Xapian::Database db(dir.toUtf8().constData());
+    Xapian::Database db;
+    try {
+        db = Xapian::Database(dir.toUtf8().constData());
+    } catch (const Xapian::DatabaseError& e) {
+        kWarning() << "Failed to open Xapian database:" << QString::fromStdString(e.get_error_string());
+        return ResultIterator();
+    }
 
     QList<Xapian::Query> m_queries;
 
-- 
1.9.1

